<html>
  <head>
    <title>Choose the network reconstruction method that's best for you</title>
    <link href='http://fonts.googleapis.com/css?family=Oswald:300|Roboto+Slab:300' rel='stylesheet' type='text/css'>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="choice_tree.css" type="text/css" />
  </head>
  
  <body>
    <div id="container">
      <a href="https://github.com/olavolav/ChooseReconstruction"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_gray_6d6d6d.png" alt="Fork me on GitHub"></a>
      <header>
        <h1>Dear neuroscientist</h1>
        <h2>choose your reconstruction method</h2>
      
        <p>Finding the right statistical tool to analyze structural connectivity can be<br/>
          tough, since there are so many out there. We can help, though!</p>
      </header>
    
      <div id='criteria'>
        <h2>Adjust the criteria to restrict your search</h2>
        <div class='spacer'>&nbsp;</div>
      
        <div id='criteria-data-types'>
          <h3>Please select what data you want the method to be able to work with:</h3>
          <ul><!-- list is generated dynamically --></ul>
        </div>
        <div id='criteria-features'>
          <h3>Please select what other properties the method should have:</h3>
          <ul><!-- list is generated dynamically --></ul>
        </div>
        <div id='filterstats'>
          Please specify your criteria.
        </div>
      </div>
    
      <ul id='choice_tree'>
        <!-- list is generated dynamically -->
      </ul>
    
      <footer>
        Note that the information of methods marked with the star symbol &#9734; has been verified by the authors of the paper shown as reference.<br/>
        <span class="method_count">There are currently 0 methods listed. (Please enable JavaScript to use this app.)</span>
        <!-- number is updated dynamically -->
        Want to add your own? Please do <a href="mailto:olav@nld.ds.mpg.de">contact us</a>.
        
        <p id="impressum_trigger">Click <a onClick="$('#impressum').show();$('#impressum_trigger').hide();">here to display legal notice (Impressum)</a>.</p>
        <p id="impressum" style="display:none">
          <strong>Impressum:</strong> (§5 Telemediengesetz)<br/>
          Dr. Olav Stetter<br/>
          Max Planck Institute for Dynamics and Self-Organization<br/>
          Am Fassberg 17, 37077 Göttingen, Germany<br/>
          E-mail: <a href="mailto:olav@nld.ds.mpg.de">olav@nld.ds.mpg.de</a><br/>
          Please note that this site is using cookies to analyze traffic, unique visitors etc.
          The information obtained is anonymous and will be deleted in regular intervals.
        </p>
        <p>Design by <a href="http://www.subvergedesign.com/">Ben Hartley</a>.</p>
      </footer>
    </div>
    
    <div id='modal'>
      <!-- modal is generated dynamically -->
    </div>
    
    <script type="text/template" id="category-template">
      <ul class="filter-btn-group">
        <% var c_name = ''; %>
        
        <% c_name = "filter-btn-yes-inactive"; %>
        <% if(value===1) { c_name = "filter-btn-yes-active"; } %>
        <li class="filter-btn <%= c_name %>" data-id='1'>&#x2713;</li>
        <% if(!data_type_filter){ %>
          <% c_name = "filter-btn-no-inactive"; %>
          <% if(value===-1) { c_name = "filter-btn-no-active"; } %>
          <li class="filter-btn <%= c_name %>" data-id='-1'>&#x2717;</li>
        <% } %>
      </ul>
      <span class='name'><%= name %> <span class='filter_tooltip_icon badge'>i</span></span>
      <div class='filter_help_text'><%= help_text %></div>
    </script>

    <script type="text/template" id="method-template">
      <h2><%= name %>
      <% if(verified_by_author) { %>
        <span class="verified_symbol">&#9734;</span>
      <% } %>
      </h2>
      <% var prop = properties; %>
      <ul class="properties">
        <% _(prop).each(function(value, name) { %>
          <li>
            <span class="properties-value">
              <% if(value===true) { %>
                <span class="yes">&#x2713;</span>
              <% } else { %>
                <span class="no">&#x2717;</span>
              <% } %>
            </span>
            <span class="properties-name"><%= name %></span>
          </li>
        <% }); %>
      </ul>
      <p class="more_info">
        More Info >
      </div>
    </script>
    
    <script type="text/template" id="filterstats-template">
      There
      <% if(nr == 1) { %>
        is <span class='badge'>1</span> method that matches
      <% } else { %>
        are <span class='badge'><%= nr %></span> methods that match
      <% } %>
      your criteria.
    </script>
    
    <script type="text/template" id="stats-template">
      There are currently <%= nr %> methods listed.
    </script>
    
    <script type="text/template" id="modal-template">
      <div class="modal_header">
        <h2>More info: <%= name %></h2>
      </div>
      <div class="modal_content">
        <p><%= description %></p>

        <h3>References:</h3>
        <ul class="modal_references">
          <% _(references).each(function(url, ref) { %>
            <li><a href="<%= url %>"><%= ref %></a></li>
          <% }); %>
        </ul>

        <% if(code_repositories) { %>
          <h3>Available open source code:</h3>
          <ul class="modal_references">
            <% _(code_repositories).each(function(url, language) { %>
              <li><a href="<%= url %>"><%= language %></a></li>
            <% }); %>
          </ul>
        <% } %>

        <% if(verified_by_author) { %>
          <div class="modal_verified_line">
            (Information verified by <%= verified_by_author %>.)
          </div>
        <% } %>

      </div>
      <div class="modal_footer">OK</div>
    </script>

    
    <script type="text/javascript" src="choice_tree.min.js"></script>    
    
    <script type="text/javascript">
      var methods = new app.ReconstructionMethodList();

      // declare reconstruction methods and their features
      methods.add( new app.ReconstructionMethod({name: "Cross-Correlation",
        properties: {
          "Works with single cell resolution": true,
          "Works with calcium fluorescence data": true,
          "Works with macro scale resolution": true,
          "Works with fMRI data": true,
          "Has been tested in simulations": true,
          "Has been applied to real data": true,
          "Standard method": true
        },
        description: 'Correlation is a standard method to quantify the statistical similarity between time series. Asymmetry in the direction is achieved by introducing a time-lag. Nevertheless, correlation does not imply causation.',
        references: {'Wikipedia article': 'http://en.wikipedia.org/wiki/Correlation_and_dependence'},
        code_repositories: {'As part of the TE-Causality package (C++)': 'https://github.com/olavolav/TE-Causality', 'Matlab function xcorr': 'http://www.mathworks.com/help/signal/ref/xcorr.html'}
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Mutual Information",
        properties: {
          "Works with single cell resolution": true,
          "Non-linear": true,
          "Model-free": true,
          "Works with spike data": true,
          "Works with macro scale resolution": true,
          "Works with fMRI data": true,
          "Has been tested in simulations": true,
          "Has been applied to real data": true,
          "Standard method": true
        },
        description: 'As the non-linear generalization of Cross-correlation, Mutual Information has become a standard estimator in the neurosciences.',
        references: {
          'Wikipedia article': 'http://en.wikipedia.org/wiki/Mutual_information',
          'Margolin, A.A., Nemenman, I., Basso, K., Wiggins, C., Stolovitzky, G., Favera, R.D., and Califano, A. (2006). ARACNE: An algorithm for the reconstruction of gene regulatory networks in a mammalian cellular context. BMC Bioinformatics 7, S7.': 'http://www.ncbi.nlm.nih.gov/pubmed/16723010',
          'Quinn, C.J., Coleman, T.P., Kiyavash, N., and Hatsopoulos, N.G. (2010). Estimating the directed information to infer causal relationships in ensemble neural spike train recordings. J Comput Neurosci 30, 17–44.': 'http://www.springerlink.com/index/10.1007/s10827-010-0247-2'},
        code_repositories: {'As part of the TE-Causality package (C++)': 'https://github.com/olavolav/TE-Causality'}
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Incremental Mutual Information",
        properties: {
          "Works with single cell resolution": true,
          "Non-linear": true,
          "Model-free": true,
          "Works with spike data": true,
          "Has been tested in simulations": true,
          "Has been applied to real data": true
        },
        description: 'Incremental Mutual Information is a modification of Mutual Information which conditions out temporal dependencies in the responses of neurons, to better disambiguate different connectivity motifs.',
        references: {
          'Singh, A., and Lesica, N.A. (2010). Incremental mutual information: A new method for characterizing the strength and dynamics of connections in neuronal circuits. PLoS Comput Biol 6, e1001035.': 'http://dx.plos.org/10.1371/journal.pcbi.1001035.g005'},
        verified_by_author: "Abhinav Singh"
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Granger Causality",
        properties: {
          "Works with single cell resolution": true,
          "Works with macro scale resolution": true,
          "Works with spike data": true,
          "Works with fMRI data": true,
          "Has been tested in simulations": true,
          "Has been applied to real data": true,
          "Standard method": true
        },
        description: 'Clive Granger described causality as improved prediction, and in that sense is Granger Causality a causal estimator in a linear auto-regressive framework.',
        references: {
          'Wikipedia article': 'http://en.wikipedia.org/wiki/Granger_causality',
          'Granger, C. (1969). Investigating causal relations by econometric models and cross-spectral methods. Econometrica 37, 424–438.': 'http://www.jstor.org/stable/1912791',
          'Cadotte, A.J., DeMarse, T.B., He, P., and Ding, M. (2008). Causal measures of structure and plasticity in simulated and living neural networks. PLoS One 3, e3355.': 'http://dx.plos.org/10.1371/journal.pone.0003355.t002',
          'Ladroue, C., Guo, S., Kendrick, K., and Feng, J. (2009). Beyond element-wise interactions: Identifying complex interactions in biological processes. PLoS One 4, e6899.': 'http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2746320/',
          'Nedungadi, A.G., Rangarajan, G., Jain, N., and Ding, M. (2009). Analyzing multiple spike trains with nonparametric Granger causality. J Comput Neurosci 27, 55–64.': 'http://link.springer.com/article/10.1007/s10827-008-0126-2'
        },
        code_repositories: {'As part of the TE-Causality package (C++)': 'https://github.com/olavolav/TE-Causality'}
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Transfer Entropy",
        properties: {
          "Non-linear": true,
          "Works with single cell resolution": true,
          "Works with macro scale resolution": true,
          "Model-free": true,
          "Works with spike data": true,
          "Works with fMRI data": true,
          "Works with EEG data": true,
          "Has been tested in simulations": true,
          "Has been applied to real data": true,
          "Standard method": true
        },
        description: 'Introduced by Thomas Schreiber in 2000, Transfer Entropy is the non-linear version of Granger Causality.',
        references: {
          'Schreiber, T. (2000). Measuring information transfer. Phys Rev Lett 85, 461–464.': 'http://prl.aps.org/abstract/PRL/v85/i2/p461_1',
          'Kaiser, A., and Schreiber, T. (2002). Information transfer in continuous processes. Physica D 166, 43–62.': 'http://apps.isiknowledge.com/full_record.do?product=WOS&search_mode=GeneralSearch&qid=1&SID=T1c9EmHIiJ92Ak5mN7j&page=1&doc=4',
          'Besserve, M., Schölkopf, B., Logothetis, N.K., and Panzeri, S. (2010). Causal relationships between frequency bands of extracellular signals in visual cortex revealed by an information theoretic analysis. J Comput Neurosci 29, 547–566.': 'http://www.ncbi.nlm.nih.gov/pubmed/20396940'
        },
        code_repositories: {'As part of the TE-Causality package (C++)': 'https://github.com/olavolav/TE-Causality', 'Trentool (Matlab toolbox)': 'http://www.trentool.de/'}
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Incremental Transfer Entropy",
        properties: {
          "Non-linear": true,
          "Model-free": true
        },
        description: 'Incremental TE is an extension of TE that identifies the inputs to a given neuron by gradually building up the statistically significant set of source neurons.',
        references: {
          'Lizier, J., and Rubinov, M. (2012). Multivariate construction of effective computational networks from observational data. Pre-print from mis.mpg.de.': 'http://www.mis.mpg.de/publications/preprints/2012/prepr2012-25.html'
        },
        verified_by_author: "Joseph Lizier"
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Generalized Transfer Entropy",
        properties: {
          "Non-linear": true,
          "Works with single cell resolution": true,
          "Works with calcium fluorescence data": true,
          "Model-free": true,
          "Works with spike data": true,
          "Has been tested in simulations": true,
          "Has been applied to real data": true
        },
        description: 'Generalized Transfer Entropy was developed for neural systems that are state-dependent, for example bursting networks. It can also allow for interactions that are faster than the recording resolution, like calcium imaging data.',
        references: {'Stetter, O., Battaglia, D., Soriano, J., and Geisel, T. (2012). Model-free reconstruction of excitatory neuronal connectivity from calcium imaging signals. PLoS Comput Biol 8, e1002653.': 'http://www.ploscompbiol.org/article/info:doi/10.1371/journal.pcbi.1002653', 'Orlandi, J.G., Stetter, O., Soriano, J., Geisel, T., and Battaglia, D. (2013). Transfer Entropy reconstruction and labeling of neuronal connections from simulated calcium imaging. arXiv 1309.4287.': 'http://arxiv.org/abs/1309.4287'},
        code_repositories: {'C++': 'https://github.com/olavolav/TE-Causality', 'MatLab': 'https://github.com/olavolav/GTE-Challenge'},
        verified_by_author: "Olav Stetter"
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Bayesian Inference (Calcium)",
        properties: {
          "Non-linear": true,
          "Works with single cell resolution": true,
          "Works with calcium fluorescence data": true,
          "Has been tested in simulations": true
        },
        description: 'Using statistical inference, Mishchencko et al. reconstruct synaptic networks from population calcium imaging data using a two step approach: First infer the spike times, then infer the parameters of an assumed Generalized Linear Model to get at the synaptic weights.',
        references: {'Mishchencko, Y., Vogelstein, J.T., and Paninski, L. (2011). A Bayesian approach for inferring neuronal connectivity from calcium fluorescent imaging data. Ann Appl Stat 5, 1229–1261.': 'http://scinetcentral.com/~mishchenko/preprint-bayes.pdf'},
        code_repositories: {'NETFIT (Matlab package)': 'http://scinetcentral.com/~mishchenko/NETFIT_package.zip'},
        verified_by_author: "Yuriy Mishchenko"
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Anatomical Reconstruction",
        properties: {
          "Works with single cell resolution": true,
          "Works with anatomical data": true,
          "Model-free": true,
          "Has been applied to real data": true
        },
        description: 'Based on EM-data of thin slices of tissue, this is a computational approach to follow axons and dendrites to get a 3D reconstruction of neuronal tissue. This results in a set of contact points between neurons with putative synapses.',
        references: {'Mishchenko, Y., and Paninski, L. (2012). A Bayesian compressed-sensing approach for reconstructing neural connectivity from subsampled anatomical data. J Comput Neurosci, In press.': 'http://www.springerlink.com/index/10.1007/s10827-012-0390-z'}
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Generalized Linear Model",
        properties: {
          "Works with single cell resolution": true,
          "Non-linear": true,
          "Works with spike data": true,
          "Has been tested in simulations": true,
          "Has been applied to real data": true
        },
        description: 'Assuming a Generalized Linear Model for the neurons and using spike data as source, synapses can be extracted from the resulting connectivity kernels.',
        references: {'Pillow, J., Shlens, J., Paninski, L., Sher, A., and Litke, A. (2008). Spatio-temporal correlations and visual signalling in a complete neuronal population. Nature 454, 995–1000.': 'http://www.nature.com/nature/journal/v454/n7207/abs/nature07140.html'},
        code_repositories: {'Matlab function': 'http://www.mathworks.com/help/stats/generalized-linear-models_btq_wnf-1.html'}
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Dynamical Cascades",
        properties: {
          "Works with single cell resolution": true,
          "Non-linear": true,
          "Works with spike data": true,
          "Has been tested in simulations": true,
          "Has been applied to real data": true
        },
        description: 'In a simple cascade model, the high-dimensional inference problem can be reduced to a smaller set of putitative connections per neuron, which can then be solved statistically.',
        references: {'Pajevic, S., and Plenz, D. (2009). Efficient network reconstruction from dynamical cascades identifies small-world topology of neuronal avalanches. PLoS Comput Biol 5, e1000271.': 'http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1000271'}
      }));
      
      methods.add( new app.ReconstructionMethod({name: "Dynamical Causal Modeling",
        properties: {
          "Non-linear": true,
          "Works with macro scale resolution": true,
          "Works with fMRI data" : true,
          "Works with EEG data": true,
          "Has been tested in simulations": true,
          "Has been applied to real data": true,
          "Standard method": true
        },
        description: 'Developed for fMRI data sets, coupling parameters are inferred in a Bayesian framework assuming a bilinear nature of the nodes in the system.',
        references: {
          'Scholarpedia article': 'http://www.scholarpedia.org/article/Dynamic_causal_modeling',
          'Friston, K.J., Harrison, L., and Penny, W. (2003). Dynamic causal modelling. NeuroImage 19, 1273–1302.': 'http://www.sciencedirect.com/science/article/pii/S1053811903002027',
          'Friston, K.J., Bastos, A., Litvak, V., Stephan, K.E., Fries, P. and Moran, R. (2012). DCM for complex-valued data: Cross-spectra, coherence and phase-delays. NeuroImage 59, 439-455.': 'http://www.sciencedirect.com/science/article/pii/S1053811911008251',
          'List of publications using DCM (Wellcome Trust Centre for Neuroimaging)': 'http://www.fil.ion.ucl.ac.uk/spm/doc/biblio/Keyword/DCM.html'
        },
        code_repositories: {'Matlab': 'http://www.fil.ion.ucl.ac.uk/spm/software/'},
        verified_by_author: "Guillaume Flandin"
      }));

      var filters = new app.FilterList();
      filters.generate_from_method_collection(methods);
      // start app
      new app.AppView(methods, filters);
    </script>

    <script type="text/javascript">
      var _gauges = _gauges || [];
      (function() {
        var t   = document.createElement('script');
        t.type  = 'text/javascript';
        t.async = true;
        t.id    = 'gauges-tracker';
        t.setAttribute('data-site-id', '52aae9b6e32bb4612d0007f7');
        t.src = '//secure.gaug.es/track.js';
        var s = document.getElementsByTagName('script')[0];
        s.parentNode.insertBefore(t, s);
      })();
    </script>

  </body>
</html>
